assets/src/edit-story/components/panels/size.js

```diff
@@ -18,101 +18,216 @@
+        evt.preventDefault();
+        evt.stopPropagation();
       }
-      return { width: newWidth, height: newHeight };
-    });
-    evt.preventDefault();
-  };
+    },
+    [lockRatio, onSetProperties, state]
+  );
   return (
     <SimplePanel
       name="size"
-      title={__('Size', 'web-stories')}
-      onSubmit={handleSubmit}
+      title={__('Size & position', 'web-stories')}
+      onSubmit={updateProperties}
     >
-      <InputGroup
-        label={__('Width', 'web-stories')}
-        value={state.width}
-        isMultiple={width === ''}
-        onChange={(value) => {
-          const ratio = width / height;
-          const newWidth =
-            isNaN(value) || value === '' ? '' : parseFloat(value);
-          setState({
-            ...state,
-            width: newWidth,
-            height:
-              height !== '' && typeof newWidth === 'number' && lockRatio
-                ? dataPixels(newWidth / ratio)
-                : height,
-          });
-        }}
-        postfix={_x('px', 'pixels, the measurement of size', 'web-stories')}
-        disabled={isFill}
-      />
-      <InputGroup
-        label={__('Height', 'web-stories')}
-        value={state.height}
-        isMultiple={height === ''}
-        onChange={(value) => {
-          const ratio = width / height;
-          const newHeight =
-            isNaN(value) || value === '' ? '' : parseFloat(value);
-          setState({
-            ...state,
-            height: newHeight,
-            width:
-              width !== '' && typeof newHeight === 'number' && lockRatio
-                ? dataPixels(newHeight * ratio)
-                : width,
-          });
-        }}
-        postfix={_x('px', 'pixels, the measurement of size', 'web-stories')}
-        disabled={isFill}
-      />
-      <InputGroup
-        type="checkbox"
-        label={__('Keep ratio', 'web-stories')}
-        value={lockRatio}
-        isMultiple={false}
-        onChange={(value) => {
-          setLockRatio(value);
-        }}
-        disabled={isFill}
-      />
+      {/** Position */}
+      <Row expand>
+        <BoxedNumeric
+          prefix={_x('X', 'The X axis', 'web-stories')}
+          value={state.x}
+          isMultiple={x === ''}
+          onChange={(value) =>
+            setState({
+              ...state,
+              x: isNaN(value) || value === '' ? '' : parseFloat(value),
+            })
+          }
+          disabled={isFill}
+        />
+        <BoxedNumeric
+          prefix={_x('Y', 'The Y axis', 'web-stories')}
+          value={state.y}
+          isMultiple={y === ''}
+          onChange={(value) =>
+            setState({
+              ...state,
+              y: isNaN(value) || value === '' ? '' : parseFloat(value),
+            })
+          }
+          disabled={isFill}
+        />
+      </Row>
+      {/** Width/height & lock ratio */}
+      <Row expand>
+        <BoxedNumeric
+          prefix={__('W', 'web-stories')}
```

Comment:
This should probably use `_x` too with a note on context - and similarly for `'H'` below.

Reply:
Very interesting. It was really uncomfortable to read hehe Nice to know that we have this kind of support. I will replace both. Thanks for the detailed explanation of this @barklund.